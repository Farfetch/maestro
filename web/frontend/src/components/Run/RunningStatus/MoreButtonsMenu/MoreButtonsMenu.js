import { MoreOutlined } from "@ant-design/icons";
import { Button, Dropdown, Menu } from "antd";

import { runLogDownloadUrl } from "../../../../lib/routes";

const MoreButtonsMenu = ({ runId }) => {
  const menu = (
    <Menu>
      <Menu.Item>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href={runLogDownloadUrl(runId)}
        >
          Download logs
        </a>
      </Menu.Item>
    </Menu>
  );

  return (
    <Dropdown key="more" overlay={menu} placement="bottomRight">
      <Button type="text" icon={<MoreOutlined style={{ fontSize: 20 }} />} />
    </Dropdown>
  );
};

export default MoreButtonsMenu;
