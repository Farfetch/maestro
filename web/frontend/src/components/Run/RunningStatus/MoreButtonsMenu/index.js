import MoreButtonsMenu from "./MoreButtonsMenu";

export default MoreButtonsMenu;
