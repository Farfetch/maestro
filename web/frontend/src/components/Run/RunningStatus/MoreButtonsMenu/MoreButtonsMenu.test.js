import { render } from "@testing-library/react";
import React from "react";

import MoreButtonsMenu from "./MoreButtonsMenu";

describe("components/Run/RunningStatus/MoreButtonsMenu", () => {
  test(`should render MoreButtonsMenu component`, async () => {
    const rendered = render(<MoreButtonsMenu runId="1-2-3" />);

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });
});
