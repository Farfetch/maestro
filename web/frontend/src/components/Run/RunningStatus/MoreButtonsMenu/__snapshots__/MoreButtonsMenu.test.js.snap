// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Run/RunningStatus/MoreButtonsMenu should render MoreButtonsMenu component 1`] = `"<div><button type=\\"button\\" class=\\"ant-btn css-dev-only-do-not-override-nnuwmp ant-btn-text ant-btn-icon-only ant-dropdown-trigger\\"><span class=\\"ant-btn-icon\\"><span role=\\"img\\" aria-label=\\"more\\" style=\\"font-size: 20px;\\" class=\\"anticon anticon-more\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"more\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></span></button></div>"`;
